playbooks/roles/homebrew/tasks/main.yml

```diff
@@ -4,6 +4,11 @@
 
 - name: Install Homebrew
   shell: ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+  homebrew_tap:
+    name:
+    - homebrew/bundle
+    - homebrew/cask
+    - homebrew/services
```

Comment:
How is this relevant to the current PR?

If they are needed for some unrelated goal, that should be a separate PR.

Reply:
These are the three "default" Homebrew taps that are widely used, documented in `brew man` and not installed by default (although `Install Homebrew casks` would do so for `homebrew/cask` but it's nicer to be explicit). 

Both homebrew/bundle and homebrew/services rely on code in Homebrew/brew so should be tapped at the same time as it is installed if you are going to set `HOMEBREW_NO_AUTO_UPDATE` (as Travis CI does). Otherwise, they end up mismatched and will fail under some circumstances (which motivated this PR).

Reply:
I still haven't figured out where/how Travis sets `HOMEBREW_NO_AUTO_UPDATE` but I presume that they do?

Reply:
[Travis does NOT disable autoupdate](https://travis-ci.org/native-api/test_travis/jobs/656387659#L74).

I believe they don't want taps preinstalled that are only needed in the minority of cases -- because they would add to update time even when they are not needed.

`homebrew/cask` is probably okay if redundant -- because [it is already preinstalled](https://travis-ci.org/native-api/test_travis/jobs/656387659#L78) for some reason. `homebrew/services` looks irrelevant to Travis -- whoever needs it is themselves responsible to make it work and can install it just like any other tap.

Reply:
Thanks for investigating! Does the Travis [Homebrew addon](https://docs.travis-ci.com/user/installing-dependencies#installing-packages-on-macos) disable auto-update when it calls `brew bundle`? ([line 72 of your job log](https://travis-ci.org/native-api/test_travis/jobs/656387659#L72), in the "Installing Homebrew Packages" fold)

Does it use the `homebrew/services` tap?

Reply:
Just to double check: `brew bundle` normally auto-updates `homebrew/brew` when it automatically taps `homebrew/bundle`?

Reply:
When you `brew services` and it automatically taps `homebrew/services`, does it auto-update the other taps?

Reply:
https://docs.travis-ci.com/user/installing-dependencies/#installing-packages-on-macos was what I was going on for the autoupdate.

> When you `brew services` and it automatically taps `homebrew/services`, does it auto-update the other taps?

Yes unless autoupdate is disabled.

> `brew bundle` normally auto-updates `homebrew/brew` when it automatically taps `homebrew/bundle`?

Yes unless autoupdate is disabled.

Reply:
I tried the following [`.travis.yml`](https://github.com/jablko/fzf/blob/0b7335b00f238edc7f5b0676e4174d4f171f9408/.travis.yml):
```YAML
language: go
os: osx
script:
  - brew bundle
  - brew install tmux
  - brew bundle
```

The first `brew bundle` didn't auto-update `homebrew/brew` and [errored](https://travis-ci.org/jablko/fzf/jobs/657501983#L114): `Unknown command: bundle` (due to the versions of `homebrew/brew` and `homebrew/bundle`, i.e. Homebrew/homebrew-bundle#646).

`brew install` [did auto-update](https://travis-ci.org/jablko/fzf/jobs/657501983#L117) Homebrew, whereupon the second `brew bundle` [succeeded](https://travis-ci.org/jablko/fzf/jobs/657501983#L1084) (or errored: `No Brewfile found`, expectedly).

Why didn't the first `brew bundle` auto-update `homebrew/brew`, like `brew install` did?

Additionally I tried the following [`.travis.yml`](https://github.com/jablko/fzf/blob/d79fbd5a7d2d687cf1431e268dad86883f217097/.travis.yml) (no `language: go` this time):
```YAML
os: osx
script:
  - brew bundle
  - brew install tmux
  - brew bundle
```

In this case the [effective environment](https://travis-ci.org/jablko/fzf/builds/657498708#L8) is `objective-c` (in the "Build system information" fold) which does disable auto-update:
- https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/objective_c.rb#L43
- https://travis-ci.org/jablko/fzf/builds/657498708#L64

Unsurprisingly in this case `brew install` [didn't auto-update](https://travis-ci.org/jablko/fzf/builds/657498708#L573) Homebrew and both `brew bundle` commands [errored](https://travis-ci.org/jablko/fzf/builds/657498708#L741): `Unknown command: bundle`.

It looks to me like only the `objective-c` environment disables auto-update and `brew bundle` currently (or at least as of the macOS images' version) doesn't auto-update `homebrew/brew`, whether auto-update is on or off?

Is there any chance this is the case, or was and has since been fixed?

Reply:
Here's [another example](https://github.com/jablko/fzf/blob/3cbc69d28597b1312143e537d1e5b2f7d6e02729/.travis.yml) (`language: go`, so auto-update is on, I believe):
```YAML
language: go
os: osx
script:
  - brew bundle
  - brew tap homebrew/bundle
  - brew bundle
```

`brew bundle` taps `homebrew/bundle` but [errors](https://travis-ci.org/jablko/fzf/jobs/657525581#L114): `Unknown command: bundle`. Then `brew tap homebrew/bundle` [auto-updates Homebrew](https://travis-ci.org/jablko/fzf/jobs/657525581#L117), whereupon `brew bundle` [succeeds](https://travis-ci.org/jablko/fzf/jobs/657525581#L874).

This PR (adding `homebrew/bundle` to the macOS images) remains valid: Since [`update: true`](https://docs.travis-ci.com/user/installing-dependencies#installing-packages-on-macos) is optional, certainly in the `objective-c` environment, mismatched versions will continue to cause the addon to fail, which is unexpected.

While adding all three [official external commands](https://docs.brew.sh/Manpage#official-external-commands) makes some sense, I agree that `homebrew/services` appears all but irrelevant in a CI setting and I'm tempted to pare this down to just the minimum solution. Thoughts?

Reply:
> While adding all three [official external commands](https://docs.brew.sh/Manpage#official-external-commands) makes some sense, I agree that `homebrew/services` appears all but irrelevant in a CI setting and I'm tempted to pare this down to just the minimum solution. Thoughts?

I think it's still useful if you e.g. want to actually start any of the applications you install (like mysql).

Reply:
Makes sense. Otherwise `brew services` would work and then occasionally and unexpectedly fail, unless you knew to pay the performance cost of `brew update` all the time.

What about `brew bundle` when auto-update is on? Any idea when/if that started auto-updating Homebrew when it taps `homebrew/bundle`?